src/NUnitFramework/TestBuilder.cs

```diff
@@ -107,16 +108,29 @@ public static WorkItem CreateWorkItem(Test test)
 
         public static WorkItem CreateWorkItem(Test test, object testObject)
         {
-            var context = new TestExecutionContext();
-            context.TestObject = testObject;
-            context.Dispatcher = new SuperSimpleDispatcher();
+            return CreateWorkItem(test, testObject, null);
+        }
 
-            return CreateWorkItem(test, context);
+        internal static WorkItem CreateWorkItem(Test test, object testObject, IDebugger debugger)
```

Comment:
The namespace betrays the fact that this class is not part of the framework (in spite of the file path). This is utility code used in NUnit's self-tests. Could you change the three added methods to public?

Reply:
Actually, could you add optional parameters rather than adding overloads since there are no backwards compatibility constraints due to this class not being shipped?

Reply:
The internal is required because the IDebugger type is internal. I can change that but I think maybe it's better to have this method internal instead of IDebugger public?
Regarding the overloads, I chose them because it felt more obvious to me than having IDebugger = null which then defaults to non-null instance but I am happy to make the change, I can also see that multitude of overloads is not very obvious either.

Reply:
Ah, right. Could you make the class internal then? This file is only included in nunit.framework.tests.csproj and nunitlite.tests.csproj.